forte/data/extractor/base_extractor.py

```diff
@@ -0,0 +1,295 @@
+                Extractors.
+            3. Use self.add fucntion to add the element into vocabulary.
+
+        Args:
+            predefined (Union[Set, List]): A set or list contain
+                elements to be added into the vocabulary.
+        """
+        for element in predefined:
+            self.add(element)
+
+    def update_vocab(self, pack: DataPack,
+                    instance: Annotation):
+        r"""Functionality: Add all elements from one instance into the
+        vocabulary. For example, when the instance is Sentence and we want
+        to add all Token from one sentence into the vocabulary, we might
+        call this function.
+
+        Overwrite instrcution:
+            1. Get all entries from one instance in the pack.
+                You probably would use pack.get fucntion to acquire
+                Entry that you need.
+            2. Get elements that are needed from entries. This process will
+                be very different for different extractors. For example,
+                you might want to get the token text from one sentence.
+                Or you mgiht want to get the tags for a squence for
+                one sentence.
+            3. Use self.add to add those element into the vocabulary.
+
+        Args:
+            pack (Datapack): The datapack that contains the current
+                instance.
+            instance (Annotation): The instance from which the
+                extractor will get elements from.
+        """
+        raise NotImplementedError()
+
+    def extract(self, pack: DataPack,
+                instance: Annotation) -> Feature:
+        r"""Functionality: Extract the feature for one instance in a pack.
+
+        Overwrite instruction:
+            1. Get all entries from one isntance in the pack.
+            2. Get elements that are needed form entries. For exmple,
+                the token text or seqence tags.
+            3. Contruct a feature and return it.
+
+        Args:
+            pack (Datapack): The datapack that contains the current
+                instance.
+            instance (Annotation): The instance from which the
+                extractor will extractor feature.
+
+        Returns:
+            Feature: a feature that contains the extracted data.
+        """
+        raise NotImplementedError()
+
+    def remove_from_pack(self, pack: DataPack,
+                instance: Annotation):
+        r"""Functionality: Remove the existing feature of the instance
+        in a pack. We might remove an attribute under an entry or remove
+        the entry itself directly, which will depend on the different
+        type of extractors.
+
+        Overwrite instruction:
+            1. Get all entries from one isntance in the pack.
+            2. Remove entries or remove some attributes of the entry. You
+                might use pack.delete_entry function or setattr to do this.
+        """
+        raise NotImplementedError()
```

Comment:
My suggestion is to use `pass` instead of `NotImplementedError`. Since users can simply don't implement this method if he pass in something like a plain text document.

Reply:
Done.